### PR TITLE
Jastrow machines added

### PR DESCRIPTION
**Pull request by gugli17**
Added jastrow WF classes (with and without symmetries).

### CODE REVIEW

File: NetKet/Machine/jastrow_spin_symm.hpp [hunk 1]
```diff
@@ -0,0 +1,410 @@
+// Copyright 2018 The Simons Foundation, Inc. - All Rights Reserved.
```
Comment (gcarleo) [inline NetKet/Machine/jastrow_spin_symm.hpp]: should be `jastrow_spin.hpp`

File: NetKet/Machine/machine.hpp [hunk 2]
```diff
@@ -71,16 +73,21 @@ class Machine : public AbstractMachine<T> {
...
+    } else if (pars["Machine"]["Name"] == "Jastrow") {
+      m_ = Ptype(new Jastrow<T>(hilbert, pars));
     }
   }
 
   void Init(const Graph &graph, const Hilbert &hilbert, const json &pars) {
     if (pars["Machine"]["Name"] == "RbmSpinSymm") {
       m_ = Ptype(new RbmSpinSymm<T>(graph, hilbert, pars));
+    } else if (pars["Machine"]["Name"] == "JastrowSymm") {
+        m_ = Ptype(new JastrowSymm<T>(graph,hilbert, pars));
```
Comment (gcarleo) [inline NetKet/Machine/machine.hpp]: some consistent indentation here and there, `clang-tidy` is the solution, see `CONTRIBUTING`

File: Test/Machine/machine_input_tests.hpp [hunk 3]
```diff
@@ -8,7 +8,12 @@ std::vector<netket::json> GetMachineInputs() {
   std::vector<netket::json> input_tests;
   netket::json pars;
 
+
```
Comment (gcarleo) [inline Test/Machine/machine_input_tests.hpp]: Saint Peter's basilica

File: Test/Machine/machine_input_tests.hpp [hunk 4]
```diff
@@ -43,5 +48,24 @@ std::vector<netket::json> GetMachineInputs() {
...
+  // Heisemberg 1d with symmetric jastrow
+  pars = {{"Graph",
+           {{"Name", "Hypercube"}, {"L", 20}, {"Dimension", 1}, {"Pbc", true}}},
+          {"Machine", {{"Name", "JastrowSymm"}}},
+          {"Hamiltonian", {{"Name", "Heisenberg"}}}};
+  input_tests.push_back(pars);
+
+
+
```
Comment (gcarleo) [inline Test/Machine/machine_input_tests.hpp]: Bernini's artwork

**Comment by kchoo1118** [comment]
Looks good.
Probably we should have some machine to add to wavefunctions or take product of wavefunctions too?

**Pull request merged**